forge-game/src/main/java/forge/game/trigger/Trigger.java

```diff
@@ -368,20 +368,23 @@ public boolean meetsRequirementsOnTriggeredObjects(Game game,  final Map<Ability
             }
         }
 
-        String condition = getParam("Condition");
+        String condition = getParamOrDefault("Condition", "");
         if ("AltCost".equals(condition)) {
             final Card moved = (Card) runParams.get(AbilityKey.Card);
             if( null != moved && !moved.isOptionalCostPaid(OptionalCost.AltCost))
                 return false;
-        } else if ("AttackedPlayerWithMostLife".equals(condition)) {
+        // currently only AttackedPlayerWithMostLife and AttackedOpponentWithMostLife
```

Comment:
no, that's wrong
you can't attack your allies anyway :P

the problem is the PlayerProperty which looks at all players